src/sentry/integrations/slack/utils.py

```diff
@@ -298,3 +308,69 @@ def get_identities_by_user(idp, users):
         status=IdentityStatus.VALID,
     )
     return {identity.user: identity for identity in identity_models}
+
+
+def get_org_member_by_slack_id(request, integration, organization, slack_id):
+    try:
+        idp = IdentityProvider.objects.get(type="slack", external_id=integration.external_id)
+    except IdentityProvider.DoesNotExist:
+        logger.error("slack.action.invalid-team-id", extra={"slack_id": integration.external_id})
+        return render_error_page(request, body_text="HTTP 403: Invalid team ID")
```

Comment:
`get_org_member_by_slack_id` sometimes returns an `OrganizationMember` and sometimes a `Response`. You'll need to pull all the logic in this method into the call site.